src/cluster.c

```diff
@@ -1284,6 +1461,38 @@ void clusterRenameNode(clusterNode *node, char *newname) {
     clusterAddNode(node);
 }
 
+void clusterAddNodeToShard(const char *shard_id, clusterNode *node) {
+    sds s = sdsnewlen(shard_id, CLUSTER_NAMELEN);
+    dictEntry *de = dictFind(server.cluster->shards,s);
+    if (de == NULL) {
+        list *l = listCreate();
+        listAddNodeTail(l, node);
+        serverAssert(dictAdd(server.cluster->shards, s, l) == DICT_OK);
+    } else {
+        list *l = dictGetVal(de);
+        if (listSearchKey(l, node) == NULL) {
+            listAddNodeTail(l, node);
+        }
+        sdsfree(s);
+    }
+}
+
+void clusterRemoveNodeFromShard(const char *shard_id, clusterNode *node) {
```

Comment:
In all cases, don't we expect `node->shard_id = shard_id`? So wouldn't the prototype of `clusterRemoveNodeFromShard(clusterNode *node);` make more sense?

Reply:
Good catch. Make sense. The mental model is that we always remove a member node and if a node is a member of a shard, its shard_id should be equal to the shard's id.